packages/server/routes/api/events.js

```diff
@@ -280,4 +281,58 @@ router.get(
   }
 );
 
+router.post(
+  "/:eventId/check-in",
+  AuthController.authenticate,
+  eventIdValidator,
+  userIdBodyValidator,
+  async (request, response) => {
+    const errors = validationResult(request);
+
+    if (!errors.isEmpty()) {
+      response.status(400).json({
+        status: "fail",
+        data: errors.array(),
+      });
+      return;
+    }
+
+    try {
+      const { status, userId } = request.body;
+      const eventId = request.params.eventId;
```

Comment:
Remember that we shouldn't access request body and parameters like this anymore now that we have `express-validator`. Import and use `matchedData(request)` from `express-validator`.

```js
const { userId, eventId } = matchedData(request);
```